### ISSUE TITLE
Unsatisfied proof if first input is public

### ISSUE THREAD

**Issue by vherbert** [scipr-lab/libiop#13]
Hi,

I test a simple R1CS with Fractal's libiop and face an unexplained behaviour. 

The R1CS is satisfied if all given inputs are auxiliary (in other words, only prover can access them, it's the witness).

In contrast, if I transform first auxiliary input into a primary input (public data), the proof does not verify fully.

That is, I obtain this message:

   * IOP transcript valid: true
  * Full protocol decision predicate satisfied: false

In other words, this configuration works:

`  const size_t num_inputs =0;
  const size_t num_variables = 7;
  const size_t num_constraints = num_variables + 1;`

This one does not:

`  const size_t num_inputs =1;
  const size_t num_variables = 7;
  const size_t num_constraints = num_variables + 1;`

Have you some suggestions? Thanks in advance.

**Comment by ValarDragon** [comment]
Hrmm, this is odd, I'll investigate this, thanks for reporting it. It should work with 1 input

**Comment by ValarDragon** [comment]
I think this is an issue with the number of constraints/variables being so low. The library doesn't have much parameterization support for the case where `max(num_constraints, num_variables) < 128`. (Essentially due to some internal details of the SNARK, there isn't a notable efficiency difference between proving/verifying circuits of size `<= 128`. 

Using 1 input works for larger sizes. (e.g. `>= 128` constraints)

Is this an important case to consider? If so, I can see what it would take to add parameterization for this (or alternatively just pad the constraint system accordingly)
